Change worker to return promise

andrewbiang888 opened the pull request:
I'm thinking we may be confusing `bull` by returning a promise AND calling `done`. The docs [here](https://optimalbits.github.io/bull/#consumers) recommend using an async function. One of the maintainers also states in [this comment](https://github.com/OptimalBits/bull/issues/1766#issuecomment-641381450) to "not use "done" and return a promise."

andrewbiang888 committed:
Change worker to return promise

andrewbiang888 committed:
fix import, rm unused import

andrewbiang888 committed:
rm empty line

jaronheard reviewed (commented):
should we catch/log an error somehow?

andrewbiang888 commented:
@jaronheard `bull` will send it to the ".on('error', (job: any, err: Error) => console.error(`error ${err}`, { job, e: renderError(err) }))" line in `jobs/queues/sendNightlyEmailQueue.ts:18:88`, the exact line that throws our current error 🙄. The catch I added (and just now removed) was not useful.

jaronheard commented:
@andrewbiang888 thanks for the clarification!
